ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetMinerDataByBlockNumberTest.java

```diff
@@ -0,0 +1,152 @@
+import static org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError.WORLD_STATE_UNAVAILABLE;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequest;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.query.BlockWithMetadata;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.api.query.TransactionWithMetadata;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.BlockHeaderTestFixture;
+import org.hyperledger.besu.ethereum.core.Difficulty;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
+import org.hyperledger.besu.ethereum.worldstate.WorldStateArchive;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Arrays;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EthGetMinerDataByBlockNumberTest {
+  @Mock private BlockchainQueries blockchainQueries;
+  @Mock private ProtocolSchedule protocolSchedule;
+  @Mock private WorldStateArchive worldStateArchive;
+  @Mock private ProtocolSpec protocolSpec;
+  @Mock private Blockchain blockChain;
+  private EthGetMinerDataByBlockNumber method;
+  private final String ETH_METHOD = "eth_getMinerDataByBlockNumber";
+  private final BlockHeaderTestFixture blockHeaderTestFixture = new BlockHeaderTestFixture();
+
+  @Before
+  public void before() {
+    this.method = new EthGetMinerDataByBlockNumber(blockchainQueries, protocolSchedule);
+  }
+
+  @Test
+  public void shouldReturnExpectedMethodNameTest() {
+    assertThat(method.getName()).isEqualTo(ETH_METHOD);
+  }
+
+  @Test
+  public void successTest() {
+    final BlockHeader header = blockHeaderTestFixture.buildHeader();
+    final BlockWithMetadata<TransactionWithMetadata, Hash> blockWithMetadata =
+        new BlockWithMetadata<>(
+            header, Collections.emptyList(), Collections.emptyList(), Difficulty.of(100L), 5);
+
+    Mockito.when(blockchainQueries.blockByNumber(Mockito.anyLong()))
```

Comment:
Optional Style: Most of our mocking uses static import of `.when()` and (when not using the junit annotations) `.mock()`.  Similarly for most of the `Assertsions.*` series to.  This tends to make a more "readable" test at the expense of making it clear this is a Mockito method.